internal/compiler/generator/rust.rs

```diff
@@ -1015,6 +1027,15 @@ fn generate_sub_component(
                 #subtree_index_function
             }
 
+            fn item_geometry(self: ::core::pin::Pin<&Self>, index: u32) -> slint::private_unstable_api::AnonymousLogicalRect {
```

Comment:
Perhaps it would be simpler to let also this function return a `sp::LogicalRect` instead of the `AnonymousLogicalRect` alias, that is otherwise unused. AFAICS there's only one call site for this function, where the Euclid rect is created. The Euclid rect creation could them be moved into here and I think then you could get rid of the AnonymousLogicalRect type alias.

Reply:
Then we'd have lots more of euclid rect creation.  This is what we do in C++.
Could do the same in C++ and rust indeed.

Reply:
Why are there more euclid rect creations? AFAICS there is only one caller for this function and it creates an Euclid rect out of the tuple.

Reply:
I meant in code size.  But it probably gets optimized away anyway.

Reply:
Last commit in the PR does that.